tripy/tripy/backend/compiler_api.py

```diff
@@ -222,6 +222,19 @@ def add(a, b):
                                 tensor,
                             ],
                         )
+            elif "InternalError: failed to set input shape" in str(err) or "Runtime shape mismatch" in str(err):
```

Comment:
Is there a better approach we can take here? Checking strings may not be scalable. Can MLIR-TRT emit a more informative error to begin with?

Reply:
It does print out information from which you can interpret what went wrong but Tripy can report better consistent errors.
Ex:
```
Unexpected tensor shape.
    For parameter `x`, expected the tensor shape `shape(32, 2)` to be within bounds for all dimensions. However, dimension 0 has a shape of 32, which is not within the expected bounds of (1, 16). Note: The provided argument was: 

    --> /tripy/tripy/frontend/trace/ops/fill.py:121 in full()
          |
      121 |     return full_impl(shape, value, dtype, output_rank)
          | 

    --> /tripy/tripy/frontend/ops/tensor_initializers.py:57 in ones()
          |
       57 |     return full(shape, 1, dtype)
          |            ^^^^^^^^^^^^^^^^^^^^^ --- required from here

    --> /tripy/tester.py:18 in <module>()
          |
       18 | inp = tp.ones((32, 2), dtype=tp.float32)
          |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ --- required from here
```
vs 
```
IExecutionContext::setInputShape: Error Code 3: API Usage Error (Parameter check failed, condition: satisfyProfile. Set dimension [32,2] for tensor arg0 does not satisfy any optimization profiles. Valid range for profile 0: [1,2]..[16,2].)
Traceback (most recent call last):
  File "/tripy/tester.py", line 19, in <module>
    print(fast_geglu(inp))
  File "/tripy/tripy/backend/compiler_api.py", line 210, in __call__
    executor_outputs = self._executor.execute(self._output_devices, input_tensors)
  File "/tripy/tripy/utils/utils.py", line 73, in wrapper
    result = func(*args, **kwargs)
  File "/tripy/tripy/backend/mlir/executor.py", line 181, in execute
    self.session.execute_function("main", in_args=in_args, out_args=out_args, stream=self.stream)
mlir_tensorrt.runtime._mlir_libs._api.MTRTException: InternalError: InternalError: failed to run function "main": [string "__check_for_function("_load_i64");..."]:35: InternalError: failed to set input shape
stack traceback:
        [C]: in function 'base._trtrt_enqueue'
        [string "__check_for_function("_load_i64");..."]:35: in function 'base.main'
```

Reply:
We could probably do this check explicilty and do not need to rely on MLIR-TRT. See `Executable.get_input_info()`, it returns the `min, max` bounds for each input.

Reply:
That's a good idea and will be robust against changes in the error message emitted by MLIR-TRT.